TRD: Implementation of DCS Chamber Status processing and DPs update

leo-barreto opened the pull request:
## Main goal
Add the processing logic of TRD chamber status and config tags data points, setup othe DCS DPs for processing.

The proposed procedure is: after a new run begins, first instances of `trd_chamberStatus` and `trd_CFGtag` get written to CCDB with run number as metadata. Following changes should be logged as alarm (first change), then warnings.

Modification and organization of other DCS DPs, such as temperatures, pressures, humidity (environment).

---

Relevant JIRA: https://alice.its.cern.ch/jira/browse/TRD-109
See previous PR for more information: https://github.com/AliceO2Group/AliceO2/pull/12079

leo-barreto committed:
TRD: implementation of DCS Chamber Status processing

leo-barreto committed:
TRD: run type DP removed from processing

leo-barreto committed:
TRD: changed alias of Env DPs, merged with Cavern DPs

leo-barreto committed:
TRD: changed default value of min number of ChamberStatus/CFGtag for updated to 522

leo-barreto committed:
TRD: removed reset of ChamberStatus/CFGtag info map after CCDB upload

leo-barreto committed:
TRD: voltage variation trigger for CCDB redefined as configurable, default value of 1V

martenole commented on Detectors/TRD/calibration/include/TRDCalibration/DCSProcessor.h:
```diff
@@ -64,33 +64,64 @@ class DCSProcessor
   const std::unordered_map<DPID, TRDDCSMinMaxMeanInfo>& getTRDCurrentsDPsInfo() const { return mTRDDCSCurrents; }
   const std::unordered_map<DPID, TRDDCSMinMaxMeanInfo>& getTRDEnvDPsInfo() const { return mTRDDCSEnv; }
   const std::unordered_map<DPID, int>& getTRDRunDPsInfo() const { return mTRDDCSRun; }
+  const std::unordered_map<DPID, int>& getTRDFedChamberStatusDPsInfo() const { return mTRDDCSFedChamberStatus; }
```
```suggestion
  const std::array<constants::MAXCHAMBER, int>& getTRDFedChamberStatusDPsInfo() const { return mTRDDCSFedChamberStatus; }
```
Probably this data type will make it easier to use the chamber status in QC and in our simulation. Then we don't need the DPID definition everywhere. The same might be a good idea for `mTRDDCSFedCFGtag`.
The other DPs will likely only be used in macros, therefore I had started with the map data type there.

martenole reviewed (dismissed):
Approving to start CI. But I believe it would be better to change the data types as commented before we enable this at P2

martenole commented on Detectors/TRD/calibration/src/DCSProcessor.cxx:
```diff
@@ -410,9 +502,106 @@ bool DCSProcessor::updateRunDPsCCDB()
+  md["runNumber"] = std::to_string(mCurrentRunNumber);
+  // TODO: define mFedStartTS and mFedEndTS, use same setup as env for now
+  o2::calibration::Utils::prepareCCDBobjectInfo(mTRDDCSFedCFGtag, mCcdbFedCFGtagDPsInfo,
+                                                "TRD/Calib/DCSDPsFedCFGtag", md, mFedCFGtagStartTS, mFedCFGtagStartTS + 3 * o2::ccdb::CcdbObjectInfo::DAY);
```
```suggestion
                                                "TRD/Calib/DCSDPsFedCFGtag", md, mFedCFGtagStartTS - 30 seconds, mFedCFGtagStartTS + 3 * o2::ccdb::CcdbObjectInfo::DAY);
```

leo-barreto committed:
TRD: changes after run test
List of changes:
  - changed type of CCDB object of chamberStatus and CFGtag from unordered maps to arrays
  - removed processing and update of trd_envTemp
  - set validity of chamberStatus and CFGtag 30 seconds before processing timestamp
  - minor changes to read DCS entries macro
  - turn off alarm/warn logic, set everything as warn

leo-barreto committed:
TRD: formatting

martenole commented:
CI failure unrelated, fullCI green, merging
